src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/RunWrapper.java

```diff
@@ -125,6 +133,40 @@ public int getNumber() throws AbortException {
         return build().getNumber();
     }
 
+    @Whitelisted
+    public JSONArray getBuildCauses() throws IOException {
+        CauseAction causeAction = build().getAction(CauseAction.class);
+
+        if (causeAction != null) {
+            StringWriter w = new StringWriter();
+            DataWriter writer = JSON.createDataWriter(causeAction, w);
+            Model<CauseAction> model = new ModelBuilder().get(CauseAction.class);
+            model.writeTo(causeAction, writer);
+                JSONObject result =  JSONObject.fromObject(w.toString());
+                // return a slightlly cleaner object
+                return result.getJSONArray("causes");
+        } else {
+            return new JSONArray();
+        }
+    }
+
+    @Whitelisted
+    public JSONArray getBuildCauses(Class<Cause> clazz) throws IOException {
```

Comment:
Regardless of the future compatibility issues with RunWrapper, I don't think we want anyone to be referring to Jenkins classes directly in their pipeline. Maybe make the parameter a `String` so users can pass the class name directly. I guess we could make this nicer for users by adding `@Symbol` annotations to `Cause` classes and then allowing users to pass those symbol names instead of the full class name.

Reply:
@mikecirioli this method doesn't seem to be needed in the UX we have defined, I propose to remove this signature for the moment.

Reply:
well you want to get causes that are a `X` or it's subclasses.  You can not do that in pipeline without a big mess.  Possibly it should not take a class but a classname or something.